src/libratbag-enums.h

```diff
@@ -327,3 +327,21 @@ enum ratbag_macro_event_type {
 	RATBAG_MACRO_EVENT_KEY_RELEASED,
 	RATBAG_MACRO_EVENT_WAIT,
 };
+
+/**
+ * @ingroup enums
+ *
+ * Device Types
+ * Describes the types specified in the .device files
+ */
+enum ratbag_device_type {
+	// Used when no DeviceType property could be found
+	TYPE_UNSPECIFIED = 0,
+	// Used for any other device type than a mouse or a keyboard
+	// for example headsets, mousepads etc.
+	TYPE_OTHER,
```

Comment:
Generally I recommend against a type "other" because it will eventually break things. Probably not a big deal for libratbag since no-one cares about minor breaks here but if, eventually, you want to add a new type `TYPE_HEADSET` then any consumer that relies on the type being `OTHER` will break.

Given that we don't appear to support any "other" device anyway I think it's better to just leave this out (of the API as well) and explicitly don't support anything that's not a mouse or keyboard. Once you have sufficient requests for other devices you'll also know better how to categorise those.

Reply:
we do use "other", there is the logitech powerplay mousepad and a headset somewhere in the data files to controll their RGB.

if there's a need to add a narrower specification, like headset, then it can be added in the future.
I don't see why code depending on "other", which there currently is none,  cannot be changed at that time.
wouldn't adding a new type need a new api version anyway?

Reply:
I also think having a `OTHER` type is a good idea. I guess it would be also possible to merge `TYPE_OTHER` and `TYPE_UNSPECIFIED`, since in the end it doesn't make that much of a difference thanks to capabilities. But on the other side, the type of a headset isn't really unspecified, it's just not one of the types where the type might be relevant.

> wouldn't adding a new type need a new api version anyway?

Yes it would, since sending an other enum than specified will break in `ratbagd.py`.

Reply:
>  wouldn't adding a new type need a new api version anyway?

The history of the version field is that we never managed to declare a stable API. The original plan was to have the API stable enough that it won't change for ages and be backwards compatible - because that may be the only way to get something other than piper support ratbagd. After all, the point of libratbag was to have a system service that does the device abstraction without being tied to any specific UI. Never managed to get to that point because ETIME. The version field was introduced to stop piper from crashing when something in ratbagd changed, it was never intended to be a long-term solution, only a hack until the API was stable enough. A decent version negotiation would allow for backwards compatibility, this wasn't planned for because declaring the API stable was always just around the corner.

Reply:
I guess when we're at the point that we can declare a stable API, we could remove `TYPE_OTHER` and categorise everything we have until then. But it would be a bummer if we have to bump for every new device that isn't a mouse or a keyboard the API. For example, maybe there will be a gamepad, or a display with RGB, etc.

Reply:
so should I change anything here? fact is we need this api change to enable differentiating between device types.

Reply:
I don't do enough work on libratbag these days to veto this, so feel free to merge with this.

Might be worth thinking about capabilities instead of types - i.e. does the device have keys, mouse-like axes, LEDs etc. Provides a more generic interface because if what we want to configure is a headset's LEDs we don't need to care that it's a headset. Likewise we don't care that something is a mouse, we care about buttons, resolutions and LEDs - the only thing that hints at a mouse is the rendering in piper but that relies on the SVG anyway.

Reply:
> Might be worth thinking about capabilities instead of types - i.e. does the device have keys, mouse-like axes, LEDs etc. Provides a more generic interface because if what we want to configure is a headset's LEDs we don't need to care that it's a headset. Likewise we don't care that something is a mouse, we care about buttons, resolutions and LEDs - the only thing that hints at a mouse is the rendering in piper but that relies on the SVG anyway.

Actually that's a good idea -  I can open an issue and put it on the roadmap.

> I don't do enough work on libratbag these days to veto this, so feel free to merge with this.

I would be fine merging this then, @FFY00 if you are fine with as well can you merge this and https://github.com/libratbag/piper/pull/536?